aunt-leahs-app/src/components/adminLocationList/locationList.jsx

```diff
@@ -0,0 +1,114 @@
+import React from 'react';
+import { List, Fab } from '@material-ui/core';
+import { LocationListItem } from './locationListItem.jsx';
+import AddIcon from '@material-ui/icons/Add';
+import CustomButton from '../customButton'
+
+export class LocationList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            locations: [
+                {
+                    name: "Tree Lot",
+                    id: 1,
+                    address: "1234 East Mall"
+                },
+                {
+                    name: "Thrift Store",
+                    id: 2,
+                    address: "1234 Robson St"
+                },
+                {
+                    name: "Fundraising Event",
+                    id: 3,
+                    address: "1234 Wesbrook Mall"
+                }],
+            newLocations: [],
+            nextId: 0
+        };
+
+        this.save = this.save.bind(this);
+        this.updateLocation = this.updateLocation.bind(this);
+        this.updateNewLocation = this.updateNewLocation.bind(this);
+        this.addNewLocation = this.addNewLocation.bind(this);
+        this.deleteLocation = this.deleteLocation.bind(this);
+        this.deleteNewLocation = this.deleteNewLocation.bind(this);
+    }
+
+    save() {
+        if (this.hasName(this.state.locations) && this.hasName(this.state.newLocations)) {
+            //save this.state.locations to db
+            //insert this.state.newLocations
+        }
+        else {
+            alert("All locations require a name!");
+        }
+    }
+
+    hasName(locations) {
+        return locations.every(location => location.name != null && location.name.length > 0);
+    }
+
+    updateLocation(id, newName) {
+        let updatedLocations = this.state.locations.slice(0);
```

Comment:
All of these local variables can be `const` instead of `let` because they are not modified

Reply:
Is there any real performance benefit or is that better practice? I change the array the next line, but I don't reassign the variable so `const` would work, but I feel like that gives the impression that it won't be changed and that isn't the case (but that could just be me 🤷‍♂).

Reply:
Yeah I still don't understand this but I changed it anyway